great_expectations/core/datasource_dict.py

```diff
@@ -92,17 +94,27 @@ def data(self) -> dict[str, FluentDatasource | BaseDatasource]:  # type: ignore[
 
         return datasources
 
-    @override
-    def __setitem__(self, name: str, ds: FluentDatasource | BaseDatasource) -> None:
+    def set_datasource(
+        self, name: str, ds: FluentDatasource | BaseDatasource
+    ) -> FluentDatasource | BaseDatasource | None:
```

Comment:
Is this an accurate return? Looks like it should be `Fluent | None`.

Instead of `None`, could we just return `Fluent | Base`? I'd love to avoid nullable returns if possible

Reply:
Good catch. Addressed in https://github.com/great-expectations/great_expectations/pull/8773/commits/db0d0ad9848fe204700b50c5cf584eda87185b1a